src/DelegationRegistry.sol

```diff
@@ -49,35 +49,38 @@ contract DelegationRegistry is IDelegationRegistry, ERC165 {
     /** 
     * See {IDelegationRegistry-delegateForAll}.
     */
-    function delegateForAll(address delegate, bool value) external override {
+    function delegateForAll(address delegate, uint256 expiry) external override {
+        require(block.timestamp < expiry || expiry == 0, "INVALID_EXPIRY");
```

Comment:
I don't think we should handle `expiry == 0` here because:
- we already have a `revokeDelegate` function
- it could be confusing to "revoke a delegate for all", but the delegate could remain as a delegate on a contract and/or a token